js/views/timestamp_view.js

```diff
@@ -56,4 +45,61 @@
             clearTimeout(this.timeout);
         }
     });
+
+    Whisper.BriefTimestampView = Whisper.TimestampView.extend({
+        getRelativeTimeSpanString: function(timestamp_) {
+            // Convert to moment timestamp if it isn't already
+            var timestamp = moment(timestamp_),
+                timediff = moment.duration(moment() - timestamp);
+
+            if (timediff.years() > 0) {
+                return timestamp.format('MMM D, YYYY');
+            } else if (timediff.months() > 0 || timediff.days() > 6) {
+                return timestamp.format('MMM D');
+            } else if (timediff.days() > 0) {
+                return timestamp.format('ddd');
+            } else if (timediff.hours() > 1 ||
+                       (timediff.hours() === 1 && timediff.minutes() >= 30)) {
+                // to match conversation view, display >= 30 minutes as a full hour
+                // if the number of hours isn't zero
+                var wrap = (timediff.minutes() >= 30 ? 1 : 0);
+                return timediff.hours() + wrap + ' hours';
+            } else if (timediff.hours() === 1 || timediff.minutes() >= 45) {
+                // to match conversation view, display >= 45 minutes as 1 hour
+                return '1 hour';
+            } else if (timediff.minutes() > 1 ||
+                       (timediff.minutes() === 1 && timediff.seconds() >= 30)) {
+                // same as hours
+                var wrap = (timediff.seconds() >= 30 ? 1 : 0)
```

Comment:
I keep forgetting that JS doesn't have block scopes. I cleaned up the wrapping logic (it wasn't clear to me what a function would do there, so I moved it to the top of the function).